src/impl/conversions.js

```diff
@@ -137,7 +137,7 @@ export function hasInvalidGregorianData(obj) {
 }
 
 export function hasInvalidTimeData(obj) {
-  const validHour = numberBetween(obj.hour, 0, 23),
```

Comment:
Hmm, does this work in general? This permits lots of other methods to be passed 24 hours, which might be OK but needs thought through

Reply:
It does indeed. After some thought, it is totally ok (and in compliance with the standard) for the hour to be 24 as long as it is parsed correctly into `00:00` for the next day. I could not find a case where parsing fails, so I believe it is all right.

I am still confused by the fact this simple change fixes the handling of midnight everywhere. Apparently it does for `DateTime`, and `Interval` both.

Reply:
Yeah, it works because the native Date constructor handles overflows in the hour position.

It has a lot of other implications:

```js
// this is a spec violation, as I understand it. 24:00 is ok but I don't think 24:30 is
// I think that's probably ok to let slide, since it's better to get more valid strings included than
// be strict about the policing bad strings
> DateTime.fromISO("2019-01-09T24:30:30").toISO()
'2019-01-10T00:30:30.000-05:00'

// this is fine and probably a good thing:
> DateTime.fromObject({ year: 1999, month: 12, day: 31, hour: 24 }).toISO()
'2000-01-01T00:00:00.000-05:00'

// This is a little wild: Nov 6, 1994 is a Sunday, which means this will complain that the day is wrong:

> DateTime.fromHTTP("Sun, 06 Nov 1994 24:00:00 GMT").invalidReason
'mismatched weekday'

//whereas Nov 6 is not Monday, but this passes because it's really parsing Monday:
> DateTime.fromHTTP("Mon, 06 Nov 1994 24:00:00 GMT").isValid
true

// (also, 24 is a spec violation for HTTP dates, which also seems minor, but still, starting to creep here)
```

Those were the only things I could find just playing with it. I think most of those are probably fine, though I may make a few changes to lock it down a bit more (it wouldn't be very hard to make it require that if the hour is 24, that the minutes, seconds, and milliseconds be zero, or to isolate the change to ISO parsing)